### DISCUSSION TITLE
Roadmap & Hello everyone

### DISCUSSION THREAD

**Discussion by x2zdigital** [Leantime/leantime#590] [General]
Hello,

Is there an official or community roadmap for this project?   Has anyone thought of setting up some type of project management?

**Comment by marcelfolaron** [comment]
Hi,

Currently no official roadmap. Unofficially I am currently working to get
ldap working and then finalize auditing and activity history.
Project Management for a project management system :D
I think we could probably start a discussion/poll here to get some
feedback. I would like to go through the backlog of bugs we have in here as
well.

Best,
Marcel

**Reply by x2zdigital** [reply]
Hi Marcel,  Thanks for the reply.  I'm about 2 hours into using Leantime and have more than a dozen UX suggestions, how should the best be added to the discussion?

**Comment by marcelfolaron** [comment]
If those are feature requests or bugs feel free to just add them as issues.

To add some color to the roadmap:
It has been difficult to find the time (I still have a full time job) to
work on large features.
In 2020/2021 I released a roadmap but was not able to complete that which
resulted in some frustration from the community.

I am committed to the project and try my best to work on the most requested
features.

**Comment by PinkieIcePie** [comment]
Hi @x2zdigital!
As Marcel has said, there isn't a roadmap, I looked as well... :) 
I spent about 2 or 3 hours trying out leantime as well and fell in love with it. Being that it is also open-source, I want to contribute towards the project adding my own features and potentially fixing some bugs too.
Discussion is good, I for one sit and read through the bugs on here trying to understand how some of them occur. 

If you have code ability, fork the repo and add your features and fixes, it's a lot easier than I realised. I am currently working on some sort of theming system for leantime and once it's fully working, I'll post about it on here, it might even do some of the UX things you're looking at? 

- Pinkie

**Comment by x2zdigital** [comment]
Hi @PinkieIcePie  
Yes it is really great, we have forked and have a running instance.  In setting up multiple projects I ran into a number of things that might work better.   

For the theming, it might be as easy as adding a blank css user file and a reference to it in content.php

I havent had a chance to go through the architecture completely so this may be totally incorrect

**Reply by x2zdigital** [reply]
@PinkieIcePie i updated my reply above

**Reply by PinkieIcePie** [reply]
That is sort of how I acheived it. I modified the original default CSS to match the theme I originally wanted and that got messy quickly and didn't give me much room to organise it, it also meant the site was the same for everyone...
My next thought was to make every element colour changeable within the administration zone via the current default/custom css but I thought this could also become very messy and again, means the theme is site-wide, not user-side. :) 
I have since modified the template files to change where the 'theme' is located and then this variable is either assigned to the user theme, or default if a theme variable cannot be found.

I am hopfully going to update my fork over the weekend with some theming stuff in it so I'll let you know :)

**Comment by x2zdigital** [comment]
@marcelfolaron  I've seeded the ideas section, sorry for the spam.   Are you going to try to do all of the code yourself or are you inviting the community?

**Comment by marcelfolaron** [comment]
Thank you! This is some great feedback! The community is invited to work on
those features.
I'm happy to help, review code and answer questions!
